### PR TITLE
Add support for using values from yaml files

### LINKED ISSUE #1700: [Helm/Release]  Load values from yaml files in addition to inline specified values
We currently only support inline specified `values` for the Helm Release resource. We should also allow specifying a list of yaml files in `valueYamlFiles` containing the values to keep parity with the Helm CLI.

### PR DESCRIPTION
**Pull request by viveklak**
### Related issues (optional)
Fixes #1700

### CODE REVIEW

File: provider/pkg/provider/helm_release.go [hunk 1]
```diff
@@ -250,11 +249,37 @@ func (r *helmReleaseProvider) getActionConfig(namespace string) (*action.Configu
...
+					return nil, fmt.Errorf("unsupported type for 'valueYamlFiles' arg: %T", v)
+				}
+			}
+		}
+	}
+
 	if err := mapstructure.Decode(stripped, &release); err != nil {
 		return nil, fmt.Errorf("decoding failure: %w", err)
 	}
+	release.Values = mergeMaps(release.Values, values)
```
Comment (viveklak) [inline provider/pkg/provider/helm_release.go:282]: The precedence - if it's not clear - will prioritize the values in the `values` field over what is specified in the files.

**Review by lblackstone — COMMENTED** [review]
Changes LGTM, but you'll also need to update the description for the `valueYamlFiles` parameter in the schema.

File: provider/pkg/provider/helm_release.go [hunk 2]
```diff
@@ -250,11 +249,37 @@ func (r *helmReleaseProvider) getActionConfig(namespace string) (*action.Configu
...
+				case *resource.Asset:
+					b, err := t.Bytes()
+					if err != nil {
+						return nil, err
+					}
+					if err = yaml.Unmarshal(b, &values); err != nil {
+						return nil, err
+					}
+				default:
+					return nil, fmt.Errorf("unsupported type for 'valueYamlFiles' arg: %T", v)
```
Comment (lblackstone) [inline provider/pkg/provider/helm_release.go:273]: Do we want to also support Archives? I just noticed that the SDK type supports both Assets and Archives.

**Comment by viveklak** [comment]
> Changes LGTM, but you'll also need to update the description for the `valueYamlFiles` parameter in the schema.

Good catch! Updated now.

File: provider/pkg/provider/helm_release.go (hunk 2 above)
Comment (viveklak) [reply provider/pkg/provider/helm_release.go:273]: Trying to keep this in-line with the helm CLI. The CLI takes yaml files and not archives. We can reconsider if we get requests from users. I can't think of an obvious use case where archives would be preferred.

**Review by lblackstone — APPROVED** [review]

**Pull request merged**